README.md

```diff
@@ -1,96 +1,119 @@
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-->
 # Examples
 
-The Example module contains all the example specific modules:
+The Examples module contains all the example specific modules:
 
 - [hr-data-generator](hr-data-generator/README.md), which can create a fake sample dataset based around possible data that a company might hold about its employees.
-- example-model, contains all the deployment agnostic example code such as the rules and example configuration, as well as all the example JVM REST based services.
-- deployment, contains all the deployment specific code and scripts
-// - [example-services](example-services/README.md), contains all the example JVM REST based services
-// - [performance](performance/README.md), which provides a performance benchmarking tool
+- example-model, contains the example JVM runners for a SpringBoot application.
+- example-library, contains all the deployment agnostic example code such as the rules, types and example configuration for prepopulating the palisade services.
+- deployment, contains all the deployment specific code and scripts.
 
 The example demonstrates different users querying an avro file over a REST api.
+The queried file is a generated HR dataset of Employee records, generated by the hr-data-generator.md.
+
+
+## HR Data Generator
+The hr-data-generator supplies a synthetic set of data, constructing a number of `Employee` objects with the following fields:
+```
+class Employee {
+    UserId uid;
+    String name;
+    String dateOfBirth;
+    PhoneNumber[] contactNumbers;
+    EmergencyContact[] emergencyContacts;
+    Address address;
+    BankDetails bankDetails;
+    String taxCode;
+    Nationality nationality;
+    Manager[] manager;
+    String hireDate;
+    Grade grade;
+    Department department;
+    int salaryAmount;
+    int salaryBonus;
+    WorkLocation workLocation;
+    Sex sex;
+}
+```
+The manager field is an array of manager, which could potentially be nested several layers deep, in the generated example manager is nested 3-5 layers deep.
+
+
+## Example Library
+
+The users, resources and policies to be used in the example are configured in the [configuration yaml](/example-library/src/main/resources/application-example.yaml).
+Each of these prepopulation vaules are loaded into the appropriate service on service start-up, see the [run-example scripts](/deployment/local-jvm/bash-scripts).
+
+In particular, the example deals with the following users:
+- User Alice has the role HR and completed the PAYROLL_TRAINING_COURSE
+- User Bob has the role ESTATES and not completed any training
+- User Eve has the role IT and not completed any training
+
+### BankDetailsRule  
+The bankDetails field should be returned:
+- if the user querying the file has completed the PAYROLL_TRAINING_COURSE and the purpose of the query is SALARY
```

Comment:
user also needs to have HR role

Reply:
Addressed in https://github.com/gchq/Palisade-examples/pull/29